homeassistant/components/smhi/config_flow.py

```diff
@@ -0,0 +1,145 @@
+"""Config flow to configure smhi component.
+
+First time the user creates the configuration and
+a location is set in the hass configuration yaml it
+will use that location and create a default
+home location. This behavior will change when
+configuration flow supports default values.
+
+When home location exists a form will show
+to add additional locations for weather data.
+
+The input location will be checked by invoking
+the API. Exception will be thrown if the location
+is not supported by the API (Swedish location only)
+"""
+from typing import Dict
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.util import slugify
+
+from .const import DOMAIN, HOME_LOCATION_NAME
+
+REQUIREMENTS = ['smhi-pkg==1.0.0']
+
+
+@callback
+def smhi_locations(hass: HomeAssistant):
+    """Return configurations of smhi component."""
+    return set((slugify(entry.data[CONF_NAME])) for
+               entry in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class SmhiFlowHandler(data_entry_flow.FlowHandler):
+    """Smhi config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize smhi forecast configuration flow."""
+        self._errors = {}
```

Comment:
Not needed

Reply:
Think I got a linter error that I used the variable before initialized it... I am going to check again

Reply:
yea i got linter error so I will keep